trade-finance-manager-ui/server/controllers/utilisation-reports/utilisation-report-reconciliation-for-report/validate-payment-details-filters.ts

```diff
@@ -0,0 +1,71 @@
+import { CURRENCY_REGEX, PaymentDetailsFilters } from '@ukef/dtfs2-common';
+import { REGEX } from '../../../constants';
+import { ErrorSummaryViewModel, PaymentDetailsFilterErrorsViewModel } from '../../../types/view-models';
+
+/**
+ * Validates if the facility ID filter is valid.
+ * @param originalUrl - The original URL of the request.
+ * @param facilityIdQuery - The facility ID query string.
+ * @returns True if the facility ID filter is valid, false otherwise.
+ */
+export const isFacilityIdFilterValid = (originalUrl: string, facilityIdQuery: string): boolean => {
+  return originalUrl.includes('paymentDetailsFacilityId') && REGEX.INT.test(facilityIdQuery) && REGEX.PARTIAL_FACILITY_ID.test(facilityIdQuery);
+};
+
+/**
+ * Validates if the payment currency filter is valid.
+ * @param originalUrl - The original URL of the request.
+ * @param paymentCurrencyQuery - The payment currency query string.
+ * @returns True if the payment currency filter is valid, false otherwise.
+ */
+export const isPaymentCurrencyFilterValid = (originalUrl: string, paymentCurrencyQuery: string): boolean => {
+  return originalUrl.includes('paymentDetailsPaymentCurrency') && CURRENCY_REGEX.test(paymentCurrencyQuery);
+};
+
+/**
+ * Validates if the payment reference filter is valid.
+ * @param originalUrl - The original URL of the request.
+ * @param paymentReferenceQuery - The payment reference query string.
+ * @returns True if the payment reference filter is valid, false otherwise.
+ */
+export const isPaymentReferenceFilterValid = (originalUrl: string, paymentReferenceQuery: string): boolean => {
+  return originalUrl.includes('paymentDetailsPaymentReference') && paymentReferenceQuery.length >= 4;
```

Comment:
What's the 4 for exactly please, can we create a constant e.g `MINIMUM_...`

Reply:
Thanks, pulled this out into a constant in `libs/common`